pan/functions.pan

```diff
@@ -161,33 +151,16 @@ function push_if = {
 
 
 ########################################################################
-##=
-## @function hostname_from_object
-## @# extracts the machine hostname from the object
-##+template name.  This assumes that your machine profile has
-##+the form profile_*.  If it does not, an error is thrown.
-## @syntax void
-## @example
-##+# "/data" will contain "myhost"
-##+object template profile_myhost.example.org;
-##+"/data" = hostname_from_object();
-##=
+#
+# hostname_from_object: extracts the hostname, without the domain,
+# from the object template name. Its name is assumed to be the host FQDN.
+#
+#
 ############################################################
 function hostname_from_object = {
     # Check cardinality.
     if (ARGC != 0) error("usage: hostname_from_object()");
 
-    # Check first for the old style profile names.
-    m = matches(OBJECT, "profile_([^\\.]+)(\\.(.*))?");
```

Comment:
shouldn't throw an error instead of just removing the code?

Reply:
SCDB doesn't support `profile_xxx` for many years and Aquilon supported it. So I don't know how it could even happen... It is enough IMO to thrown an error if the hostname is not a FQDN...